src/LineBot/Messages/Template/ImageCarouselTemplate.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright 2017-2018 Dirk Lemstra (https://github.com/dlemstra/line-bot-sdk-dotnet)
+//
+// Dirk Lemstra licenses this file to you under the Apache License,
+// version 2.0 (the "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at:
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations
+// under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Newtonsoft.Json;
+
+namespace Line
+{
+    /// <summary>
+    /// Encapsulates a image carousel template.
+    /// </summary>
+    public sealed class ImageCarouselTemplate : IImageCarouselTemplate
+    {
+        private IEnumerable<IImageCarouselColumn> _columns;
+
+        [JsonProperty("type")]
```

Comment:
Can you put this inside a pragma block to fix the stylecop compile error? `#pragma warning disable 0414 // Suppress value is never used.` should be added the same was this is done inside `CarouselTemplate`.